src/Microsoft.Extensions.Configuration.AzureAppConfiguration/AzureAppConfigurationProvider.cs

```diff
@@ -279,15 +282,41 @@ public void ProcessPushNotification(PushNotification pushNotification, TimeSpan?
                     $"{nameof(pushNotification)}.{nameof(pushNotification.ResourceUri)}");
             }
 
-            _client.UpdateSyncToken(pushNotification.SyncToken);
+            if (_configClientManager.UpdateSyncToken(pushNotification.ResourceUri, pushNotification.SyncToken))
+            {
+                SetDirty(maxDelay);

```

Comment:
How do we make sure the next refreshing is using the client that matches the endpoint with notification?

Reply:
We don't.
We had discussed offline to just refresh when the notification is received. The clients may or may not get the latest values.

Reply:
All right, unless we introduce the LKG concept, it's hard to prioritize the next refresh to use the endpoint that received the notification in the current design. Can we open an issue to track this?